snap/component.go

```diff
@@ -62,6 +62,11 @@ func NewComponentSideInfo(cref naming.ComponentRef, rev Revision) *ComponentSide
 	}
 }
 
+// Equals compares two ComponentSideInfo
+func (csi *ComponentSideInfo) Equals(other *ComponentSideInfo) bool {
```

Comment:
I think we tend to use "Equal" more than "Equals"  for this kind of methods. golang is not very consistent on this. This needs a quick unit test